src/core/events/bpf.rs

```diff
@@ -152,44 +205,69 @@ impl EventFactory for BpfEventsFactory {
             0
         };
 
-        // Finally make our ring buffer and associate our map to our event
-        // processing closure.
-        let mut rb = libbpf_rs::RingBufferBuilder::new();
-        rb.add(&self.map, process_event)?;
-        let rb = rb.build()?;
-        let rs = self.run_state.clone();
-        // Start an event polling thread.
-        self.handle = Some(thread::spawn(move || {
-            while rs.running() {
-                if let Err(e) = rb.poll(Duration::from_millis(BPF_EVENTS_POLL_TIMEOUT_MS)) {
-                    match e.kind() {
-                        // Received EINTR while polling the
-                        // ringbuffer. This could normally be
-                        // triggered by an actual interruption
-                        // (signal) or artificially from the
-                        // callback. Exit without printing any error.
-                        libbpf_rs::ErrorKind::Interrupted => (),
-                        _ => error!("Unexpected error while polling ({e})"),
-                    }
-                    break;
+        let run_state = self.run_state.clone();
+        let process_log = move |data: &[u8]| -> i32 {
+            if data.len() != mem::size_of::<LogEvent>() {
+                error!("Unexpected log event size");
+                return 0;
+            }
+            // If a termination signal got received, return (EINTR)
+            // from the callback in order to trigger the event thread
+            // termination. This is useful in the case we're
+            // processing a huge number of buffers and rb.poll() never
+            // times out.
+            if !run_state.running() {
+                return -4;
+            }
+
+            let mut log_event = LogEvent::default();
+            if plain::copy_from_bytes(&mut log_event, data).is_err() {
```

Comment:
Please use `if let Err(e) ...` so the error can be shown in the log msg.

Reply:
plain::error doesn't implement Display. Are you suggesting to use Debug?

Reply:
Ah right. Up to you then, to keep it as-is or use Debug.

Reply:
I think we should be fine with `Debug`